### DISCUSSION TITLE
persistQueryClient removes cached entries if the network is down

### DISCUSSION THREAD

**Discussion by gabrielbull** [TanStack/query#2207] [Ideas]
Hi @tannerlinsley. This is sort of a follow up on my recent PR #2131. Currently, entries are removed from persisted cache when an error happens on a query, even a network error. One of the most common use case for persistQueryClient is to have offline support, which means all queries would have network errors. Using the PR I made, we can now pass dehydrateOptions and avoid the cache removing entries with code like this:

```ts
persistQueryClient({
  queryClient,
  persistor,
  dehydrateOptions: {
    shouldDehydrateQuery: () => true
  }
})
```

This is great and we can now effectively use persistQueryClient for offline support. But this begs the question, shouldn't something like that be the **default** behaviour for persistQueryClient. Why does persistQueryClient need to remove cached entries on errors anyway?

@makirby, can you tell us what was your use case for passing dehydrate/hydrate options to persistQueryClient?

**Comment by bombillazo** [comment]
Hey @gabrielbull , thanks for this, is this already working? 

I'm trying to use RQ as well for offline mode but once the device goes offline and the data gets stale, or when RQ tries to fetch data, or clears the data internally at some point idk, my whole app stops working offline due to `network request failed` errors. 😦 

I tried this approach but still the cache is cleared or invalidated or unused.

**Reply by gabrielbull** [reply]
The example shown above should work if you have the version of RQ that has the PR mentioned above.
